src/JsonService.ts

```diff
@@ -99,22 +99,27 @@ export class JsonService {
         if (contentType && !this._contentTypes.find(item => contentType.startsWith(item))) {
             throw new Error(`Invalid response Content-Type: ${(contentType ?? "undefined")}, from URL: ${url}`);
         }
-        let json: Record<string, unknown>;
-        try {
-            json = await response.json();
-        }
-        catch (err) {
-            this._logger.error("postForm: Error parsing JSON response", err);
-            if (response.ok) throw err;
-            throw new Error(`${response.statusText} (${response.status})`);
+
+        let json: Record<string, unknown> = {};
+        if (hasResponseData) {
+            try {
+                json = await response.json();
+            }
+            catch (err) {
```

Comment:
I'm not a fan of this new parameter, I think this function should be refactored to optimistically parse a non-empty response body instead. Something like this:

```suggestion
        const body = await response.text();
        let json: Record<string, unknown> = {};
        if (body) {
          try {
            json = JSON.parse(body);
          }
          catch { ... }
        }
```

Reply:
good point, i will adapt, should we add a try catch around response.text() or can't that happen?

Reply:
I'm not aware of a situation where it can throw outside of poor network conditions